docs/hd.md

```diff
@@ -0,0 +1,124 @@
+# Cardano SL HD Wallets
```

Comment:
We should include a section called `Glossary` where we clarify terms which might be ambiguous or non-obvious.

For example, I would start by writing something like:

```
## Glossary
Public Key: Explains what a Public Key is.
Secret Key: Say that it corresponds to (essentially) a Private key.
Extended Public Key:
Extended Private Key:
...
```